File: src/frontend/static/js/swal.js
const baseConfig = {
    allowOutsideClick: true,
    scrollbarPadding: false,
    heightAuto: false,
    showClass: {
        popup: "swal2-show",
        backdrop: "swal2-backdrop-show",
        icon: "swal2-icon-show",
    },
    hideClass: {
        popup: "swal2-hide",
        backdrop: "swal2-backdrop-hide",
        icon: "swal2-icon-hide",
    },
    customClass: {
        container: "modal-open",
    },
};

if (typeof Swal !== "undefined") {
    Swal.mixin({
        scrollbarPadding: false,
        heightAuto: false,
    });
}

const SwalUtils = {
    success: function (message, title = "Sucesso") {
        return Swal.fire({
            ...baseConfig,
            icon: "success",
            title: title,
            text: message,
            confirmButtonColor: "#89e67e",
        });
    },

    error: function (message, title = "Erro") {
        return Swal.fire({
            ...baseConfig,
            icon: "error",
            title: title,
            text: message,
            confirmButtonColor: "#e66360",
        });
    },

    confirm: function (message, title = "Confirmação", confirmText = "Sim", cancelText = "Não") {
        return Swal.fire({
            ...baseConfig,
            icon: "question",
            title: title,
            text: message,
            showCancelButton: true,
            confirmButtonText: confirmText,
            cancelButtonText: cancelText,
            confirmButtonColor: "#6ea2eb",
            cancelButtonColor: "#666d74",
            reverseButtons: true,
        });
    },

    warning: function (message, title = "Atenção") {
        return Swal.fire({
            ...baseConfig,
            icon: "warning",
            title: title,
            text: message,
            confirmButtonColor: "#e6714b",
        });
    },

    toast: function (message, icon = "success", position = "top-end") {
        return Swal.fire({
            toast: true,
            position: position,
            icon: icon,
            title: message,
            showConfirmButton: false,
            timer: 3000,
            timerProgressBar: true,
            showClass: {
                popup: "swal2-show",
                backdrop: "swal2-backdrop-show",
            },
            hideClass: {
                popup: "swal2-hide",
                backdrop: "swal2-backdrop-hide",
            },
        });
    },

    loading: function (message = "Carregando...") {
        return Swal.fire({
            ...baseConfig,
            title: message,
            allowOutsideClick: false,
            showConfirmButton: false,
            willOpen: function () {
                Swal.showLoading();
            },
        });
    },

    prompt: function (message, title = "Digite", inputPlaceholder = "") {
        return Swal.fire({
            ...baseConfig,
            title: title,
            text: message,
            input: "text",
            inputPlaceholder: inputPlaceholder,
            showCancelButton: true,
            confirmButtonText: "OK",
            cancelButtonText: "Cancelar",
            inputValidator: function (value) {
                if (!value) {
                    return "Você precisa escrever algo!";
                }
            },
        });
    },

    html: function (htmlContent, title = "") {
        return Swal.fire({
            ...baseConfig,
            title: title,
            html: htmlContent,
            confirmButtonText: "OK",
        });
    },

    delete: function (message = "Esta ação não poderá ser revertida!", title = "Tem certeza?") {
        return Swal.fire({
            ...baseConfig,
            title: title,
            text: message,
            icon: "warning",
            showCancelButton: true,
            confirmButtonColor: "#e66360",
            cancelButtonColor: "#666d74",
            confirmButtonText: "Sim, deletar!",
            cancelButtonText: "Cancelar",
        });
    },

    autoClose: function (message, title = "", timer = 2000) {
        return Swal.fire({
            ...baseConfig,
            title: title,
            text: message,
            timer: timer,
            timerProgressBar: true,
            showConfirmButton: false,
        });
    },
};

File: src/backend/swal.py
"""Sweetalert2 wrapper functions for PyWebview"""

import json


def _execute_swal(window, function_name, *args):
    args_str = json.dumps(args)[1:-1]
    result = window.evaluate_js(f'SwalUtils.{function_name}({args_str})')
    return result


def success(
    window,
    message,
    title='SUCESSO',
):
    return _execute_swal(
        window,
        'success',
        message,
        title,
    )


def error(
    window,
    message,
    title='UM ERRO OCORREU',
):
    return _execute_swal(
        window,
        'error',
        message,
        title,
    )


def confirm(
    window,
    message,
    title='CONFIRMAÇÃO',
    confirm_text='Sim',
    cancel_text='Não',
):
    return _execute_swal(
        window,
        'confirm',
        message,
        title,
        confirm_text,
        cancel_text,
    )


def warning(
    window,
    message,
    title='ATENÇÃO',
):
    return _execute_swal(
        window,
        'warning',
        message,
        title,
    )


def toast(
    window,
    message,
    icon='success',
    position='top-end',
):
    return _execute_swal(
        window,
        'toast',
        message,
        icon,
        position,
    )


def loading(
    window,
    message='Carregando...',
):
    return _execute_swal(
        window,
        'loading',
        message,
    )


def prompt(
    window,
    message,
    title='Digite',
    input_placeholder='',
):
    return _execute_swal(
        window,
        'prompt',
        message,
        title,
        input_placeholder,
    )


def html(
    window,
    html_content,
    title='',
):
    return _execute_swal(
        window,
        'html',
        html_content,
        title,
    )


def delete(
    window,
    message='Esta ação não poderá ser revertida!',
    title='Tem certeza?',
):
    return _execute_swal(
        window,
        'delete',
        message,
        title,
    )


def auto_close(
    window,
    message,
    title='',
    timer=2000,
):
    return _execute_swal(
        window,
        'autoClose',
        message,
        title,
        timer,
    )


def close(window, self):
    window.evaluate_js('Swal.close()')
